### PR TITLE
JS: Implement new metric queries for line counting

### PR DESCRIPTION
**Pull request by asgerf**
JS version of https://github.com/github/codeql/pull/5271.

### CODE REVIEW

File: javascript/ql/src/Summary/LinesOfCode.ql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @id js/summary/lines-of-code
+ * @name Total lines of JavaScript and TypeScript code in the database
```
Comment (esbena) [inline javascript/ql/src/Summary/LinesOfCode.ql:3]: Nit: Should this be hedged to account for the "technically-not-JavaScript" code from Angular and similar related languages that we handle? (Given the debugging purpose of this query, I think it is fine to count a few more lines than what the name promises.)

**Review by esbena — APPROVED** [review]
LGTM. See minor naming concern.

File: javascript/ql/src/Summary/LinesOfCode.ql (hunk 1 above)
Comment (asgerf) [reply javascript/ql/src/Summary/LinesOfCode.ql:3]: I'm more worried about us overthinking this and coming up with a confusing name that nobody understands. I'd rather keep it like this.

Comment (esbena) [reply javascript/ql/src/Summary/LinesOfCode.ql:3]: Ok. You have the approval already.

**Pull request merged**